core/src/main/java/hudson/tasks/Maven.java

```diff
@@ -155,8 +168,12 @@ public Maven(String targets,String name, String pom, String properties, String j
         this(targets, name, pom, properties, jvmOptions, usePrivateRepository, null, null);
     }
     
-    @DataBoundConstructor
     public Maven(String targets,String name, String pom, String properties, String jvmOptions, boolean usePrivateRepository, SettingsProvider settings, GlobalSettingsProvider globalSettings) {
+        this(targets, name, pom, properties, jvmOptions, usePrivateRepository, settings, globalSettings, true);
```

Comment:
or make setter and use it from this constructor, then there will be 2 constructors: 1 with default, 2 with setting